src/sign.c

```diff
@@ -373,30 +373,35 @@ int wally_ec_sig_verify(const unsigned char *pub_key, size_t pub_key_len,
                         const unsigned char *sig, size_t sig_len)
 {
     secp256k1_pubkey pub;
+    secp256k1_xonly_pubkey xonly_pub;
     secp256k1_ecdsa_signature sig_secp;
     const secp256k1_context *ctx = secp_ctx();
     bool ok;
 
-    if (!pub_key || pub_key_len != EC_PUBLIC_KEY_LEN ||
-        !bytes || bytes_len != EC_MESSAGE_HASH_LEN ||
+    if (!pub_key || (pub_key_len != EC_PUBLIC_KEY_LEN &&
+        pub_key_len != EC_XONLY_PUBLIC_KEY_LEN))
+        return WALLY_EINVAL;
+
```

Comment:
You check these lengths below for each case as well so you can remove this whole `if` and just add pub_key to the checks immediately below I reckon.